services/src/main/java/org/exoplatform/addons/gamification/connector/RuleIndexingServiceConnector.java

```diff
@@ -119,4 +120,9 @@ private String getUserIdentityId(String username) {
   private String toMilliSecondsString(Date date) {
     return date != null ? String.valueOf(date.getTime()) : "0";
   }
+
+  private String getDescription(String s) {
+    s = StringEscapeUtils.unescapeHtml(s);
+    return s.replace("<", " <").replace(">", "> ");
```

Comment:
why this change ?

Reply:
we need to escape the Unicode from the description and put a space between the HTML tags and the string if they are concatenated they will not be searched

Reply:
We may have to add a new analyzer such as https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-htmlstrip-charfilter.html#analysis-htmlstrip-charfilter-analyzer-ex that will include `html_strip` as `char_filter` in the default list of analyzers https://github.com/Meeds-io/commons/blob/develop/commons-search/src/main/resources/es-default-index-settings.json#L23-L30, WDYT ?

Reply:
to be checked!